src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSources/DocumentDiagnosticSource.cs

```diff
@@ -10,11 +10,11 @@
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics;
 
 internal sealed class DocumentDiagnosticSource
-    : AbstractDocumentDiagnosticSource<Document>
+    : AbstractDocumentDiagnosticSource<TextDocument>
```

Comment:
why do we need to modify our c# diagnostics handler here?

Reply:
To allow diagnostics coming from Additionalfiles

Reply:
I'm torn on whether a roslyn handler should report the additional file diagnostics or if there should be a xaml handler.  I guess actually for workspace diagnostics, we would have to report the additiona file diagnostics (since we report everything).  So perhaps it makes sense for roslyn to just report them....

Reply:
This works rather nicely actually. All anyone needs to do is write an analyzer and codefixer for additional files and Roslyn takes care of the rest. Since additional files are a Roslyn concept, I think it does make sense to support it this way. In fact, I would consider it a bug that it doesn't already do it. :)

Reply:
Yeah don't doing this.